### PR TITLE
Small updates

### PR DESCRIPTION
**Pull request by abelsiqueira**
Three small updates related to #17.
- Better handling of types
- Check length
- Add subsolver_logger

### CODE REVIEW

File: src/al.jl [hunk 1]
```diff
@@ -41,11 +45,15 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
...
   Jx = jac(nlp, x)
   fx, gx = objgrad(nlp, x)
+  lvar = eltype(nlp.meta.lvar) == T ? nlp.meta.lvar : T.(nlp.meta.lvar)
+  uvar = eltype(nlp.meta.uvar) == T ? nlp.meta.uvar : T.(nlp.meta.uvar)
 
   # penalty parameter
   μ = T.(10.0)
   # Lagrange multiplier
-  y = T.(cgls(Jx', gx)[1])
+  y = with_logger(subsolver_logger) do
```
Comment (dpo) [inline src/al.jl:54]: Why not use the µ stored in the model two lines above?

File: src/al.jl [hunk 2]
```diff
@@ -76,24 +84,24 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
...
+    S = with_logger(subsolver_logger) do
       tron(al_nlp, x = copy(al_nlp.x))
     end
 
     normcx = norm(al_nlp.cx)
 
     if normcx <= eta
       update_y!(al_nlp)
-      eta = T.(eta / (al_nlp.mu)^0.9)
+      eta = eta / al_nlp.mu^T(0.9)
```
Comment (dpo) [inline src/al.jl:95]: Greek letters?

File: src/al.jl [hunk 3]
```diff
@@ -76,24 +84,24 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
...
     else
       μ = 100 * μ
       update_mu!(al_nlp, μ)
-      eta = T.(1 / μ^0.1)
+      eta = 1 / μ^T(0.1)
```
Comment (dpo) [inline src/al.jl:99]: µ vs mu

File: src/al.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/al.jl:54]: This is the starting `mu`.

File: src/al.jl (hunk 2 above)
Comment (abelsiqueira) [reply src/al.jl:95]: I want to make the complete change to greek on a separate PR

File: src/al.jl (hunk 1 above)
Comment (dpo) [reply src/al.jl:54]: Ah ok, the AL model hasn't been created yet. But shouldn't µ be a parameter?

Comment (abelsiqueira) [reply src/al.jl:54]: You're right, since we're using `al_nlp.x` and `al_nlp.cx`, it makes sense to use `al_nlp.mu`.

File: src/al.jl [hunk 4]
```diff
@@ -58,7 +56,7 @@ function al(nlp :: AbstractNLPModel;
   eta = 0.5
 
   # create initial subproblem
-  al_nlp = AugLagModel(nlp, y, μ, x, cons(nlp, x))
+  al_nlp = AugLagModel(nlp, y, T(10.0), x, cons(nlp, x))
```
Comment (dpo) [inline src/al.jl:60]: As a parameter to the solver though, no?!

Comment (abelsiqueira) [reply src/al.jl:60]: Makes even more sense.

**Pull request merged**